### ISSUE TITLE
No events are recorded (events tab always empty)

### ISSUE THREAD

**Issue by JuDelCo** [silexphp/Silex-WebProfiler#40]
This bug has been present for a long time ago I think. The problem is the profiler is not recording a single event at all. Always yelling:

 `No events have been recorded. Are you sure that debugging is enabled in the kernel?`

I went freak yesterday (I wanted to fix this) and I spent 10 hours trying to understand the kernel event system and to track, understand & fix this bug.

---

**The problem**:

This bug is caused when $app['data_collectors'] is created, in the events line 76:

``` php
'events'    => $app->share(function ($app) { return new EventDataCollector(); }),
```

`EventDataCollector` NEEDS the dispatcher as a argument to collect the events, but here is not passed!  So, I tried to fix this passing the argument:

``` php
'events'    => $app->share(function ($app) { return new EventDataCollector($app['dispatcher']); }),
```

... and then all went **wrong**. Server crashed everytime, no error messages at all. Dafuq, why???  D:

Well, this is because in order to instantiate the new extended `$app['dispatcher']` service to pass it as an argument to `EventDataCollector`, it needs to instantiate too a `TraceableEventDispatcher`, who needs the `$app['stopwatch']` service, BUT that service hasn't been created yet ! (is more below in the code, line 149). Not only that, the service needs too to instantiate the `$app['profiler']` service, who needs to use the `$app['profiler.storage']` variable (which hasnt been set !.... and it needs the `$app['profiler.cache_dir']` which is passed by argument in the configuration files, not in this source), but not only that, even the `$app['profiler']` service has not been defined yet (is in the line 120, AND it needs to use `$app['data_collectors']` variable xDDD).

**Tl;Dr**: There are a hell of dependencies to successfully instantiate the `EventDataCollector` the right way because it needs the dispatcher service as argument and is NOT ready to that yet.

---

**Proposed solutions**:
- Put the line I suggest below as a temporally fix in the `boot` function, because dispatcher is instantiated there.
- Move all the data_collectors sets to profiler service to the `boot` function below (and fix the `EventDataCollector` call passing it the dispatcher).
- Update `EventDataCollector` to make it not neccesary to pass it as an argument the dispatcher service (using a function to set/update it maybe?).
- Forget about the events profiling xDDD

---

**Temporally fix for users**:

Meanwhile, I suggest you to add this in your development configuration file (or where you register the web profiler service) to temporally fix this bug:

``` php
// This is as always ...
$app->register(new WebProfilerServiceProvider(), array(
    'profiler.cache_dir'    => __DIR__.'/../var/cache/profiler',
    'profiler.mount_prefix' => '/_profiler'
));

// THE FIX !
$app['profiler']->add(new \Symfony\Component\HttpKernel\DataCollector\EventDataCollector($app['dispatcher']));
```

Thanks fabien for Silex, it's awesome  ;)
